inttests/volume_group_test.go

```diff
@@ -0,0 +1,388 @@
+	// Delete all the volumes from the volume group.
+	err := deleteAllVolumesInVG(s.client, s.vg.this.ID, s.vg.volumeIDs)
+	assert.NoError(s.T(), err)
+
+	// Delete that volume group, retaining the volume group ID.
+	_, err = s.client.DeleteVolumeGroup(context.Background(), s.vg.this.ID)
+	assert.NoError(s.T(), err)
+
+	// Try to configure metro volume group using the deleted vg ID.
+	_, err = s.client.ConfigureMetroVolumeGroup(context.Background(), s.vg.this.ID, &s.metro.config)
+
+	if assert.Error(s.T(), err) {
+		assert.Equal(s.T(), http.StatusNotFound, err.(g.APIError).StatusCode)
+		assert.Contains(s.T(), err.(g.APIError).Message, "Unable to find volume group")
+	}
+}
+
+// Execute ConfigureMetroVolume with a bad request body.
+func (s *MetroVolumeGroupTestSuite) TestMetroVGBadRequest() {
+	// Pass an emtpy configuration body with the request.
+	_, err := s.client.ConfigureMetroVolumeGroup(context.Background(), s.vg.this.ID, nil)
+
+	if assert.Error(s.T(), err) {
+		assert.Equal(s.T(), http.StatusBadRequest, err.(g.APIError).StatusCode)
+	}
+}
+
+/*
+/	////////////////////////////////
+/	/ END METRO VOLUME GROUP TESTS /
+/	////////////////////////////////
+*/
+type EndMetroVolumeGroupTestSuite struct {
+	suite.Suite
+
+	MetroVolumeGroupTest
+}
+
+func TestEndMetroVolumeGroupSuite(t *testing.T) {
+	suite.Run(t, new(EndMetroVolumeGroupTestSuite))
+}
+
+func (s *EndMetroVolumeGroupTestSuite) SetupSuite() {
+	// Get a new client.
+	s.client = GetNewClient()
+
+	// Get the remote PowerStore array.
+	remoteSystem := GetRemoteSystemForMetro(s.client, s.T())
+	if remoteSystem.ID == "" {
+		s.T().Skip("Could not get a remote system configured for metro. Skipping test suite...")
+	}
+
+	s.metro.config = g.MetroConfig{RemoteSystemID: remoteSystem.ID}
+
+	// Make sure remote VGs are always deleted.
+	s.metro.endOpts = g.EndMetroVolumeGroupOptions{
+		DeleteRemoteVolumeGroup: true,
+	}
+}
+
+func (s *EndMetroVolumeGroupTestSuite) SetupTest() {
+	// Create a volume to add to the vg to make it a valid vg we can test with.
+	volID, _ := CreateVol(s.T())
+	s.vg.volumeIDs = append(s.vg.volumeIDs, volID)
+
+	// Create a unique vg name for each test run.
+	s.vg.this.Name = VGPrefix + randString(8)
+
+	// Create a volume group to run tests against.
+	resp, err := s.client.CreateVolumeGroup(context.Background(), &g.VolumeGroupCreate{
+		Name:                   s.vg.this.Name,
+		VolumeIDs:              s.vg.volumeIDs,
+		IsWriteOrderConsistent: true,
+	})
+	assert.NoError(s.T(), err)
+
+	s.vg.this.ID = resp.ID
+
+	// Create a metro vg session.
+	_, err = s.client.ConfigureMetroVolumeGroup(context.Background(), s.vg.this.ID, &s.metro.config)
+	if err != nil {
+		s.T().Skipf("Could not create metro volume group session. Skipping test... Err: %s", err)
+	}
+}
+
+func (s *EndMetroVolumeGroupTestSuite) TearDownTest() {
+	// End the metro session if one still exists.
+	s.client.EndMetroVolumeGroup(context.Background(), s.vg.this.ID, &s.metro.endOpts)
```

Comment:
Won't we get the below error for successful end-metro test (_TestEndMetroVolumeGroup_())?
"the volume group is not in a metro replication session"

Reply:
Yes. That's why I've chosen not to bother checking the error response. We don't care if it fails here, we just want to clean up the PowerStore array and as a precursor to deleting the volumes and volume groups, we have to make sure it isn't still part of a replication session.